app/src/main/java/com/springboot/webpush/controller/SubscriptionController.java

```diff
@@ -26,4 +33,16 @@ public void unsubscribe(@RequestBody WebPushSubscription subscription) {
 		subscriptionService.unsubscribe(subscription);
 	}
 
+	@GetMapping("/api/vapid-public")
+	public String vapidPublic() {
+		/*
+		 * PublicKey publicKey = this.notificationService.getPublicKey();
+		 * System.out.println(publicKey);
+		 * byte[] encodedPublicKey = publicKey.getEncoded();
+		 * String b64PublicKey = Base64.getEncoder().encodeToString(encodedPublicKey);
+		 * return b64PublicKey;
+		 */
+		return "BJ3kKgvZR34Wt-dJMG9L7Lh03ISRlWueIgg-Vp5V5oYsDPr47_szfNsxxLQrghPtuN3XaI6A3FVZVZuVa7LZOFY";
```

Comment:
This is where I hardcoded the vapid public key. Should try to get it working with the standard key generation. I suspect it didn't work because maybe we double-encoded the key